### PR TITLE
Add ipc command `podfile_json` converts a Podfile to JSON

### PR DESCRIPTION
**Pull request by dacaiguoguogmail**

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! 
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* add ipc command converts a Podfile to Json 
  [dacaiguoguogmail](https://github.com/dacaiguoguogmail)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: lib/cocoapods/command/ipc/podfile_json.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+module Pod
+  class Command
+    class IPC < Command
+      class PodfileJson < IPC
+        include ProjectDirectory
+
+        self.summary = 'Converts a Podfile to Json'
```
Comment (orta) [inline lib/cocoapods/command/ipc/podfile_json.rb]: `Json` -> `JSON` please, and below

File: CHANGELOG.md [hunk 2]
```diff
@@ -32,6 +32,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
   [Austin Emmons](https://github.com/atreat)
   [#6742](https://github.com/CocoaPods/CocoaPods/issues/6742) 
 
+* Add ipc command `podfile_json` converts a Podfile to Json
```
Comment (segiddins) [inline CHANGELOG.md]: please add a period and 2 spaces to the changelog entry, in addition to capitalizing `JSON`

**Comment by dacaiguoguogmail** [comment]
@segiddins can you give me some tips on `spec/functional/command/ipc/podfile_json_spec.rb`
My test is failed
`converts a Podfile to JSON and prints it to STDOUT [FAILED]`

**Comment by dnkoutso** [comment]
@dacaiguoguogmail test failure.

File: CHANGELOG.md [hunk 3]
```diff
@@ -32,6 +32,11 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
   [Austin Emmons](https://github.com/atreat)
   [#6742](https://github.com/CocoaPods/CocoaPods/issues/6742) 
 
+* Add ipc command `podfile_json` converts a Podfile to JSON
+  [Dacaiguoguo](https://github.com/dacaiguoguogmail)
+  [#6779](https://github.com/CocoaPods/CocoaPods/pull/6779)
+
+
```
Comment (dnkoutso) [inline CHANGELOG.md]: nit extra line here.

**Comment by dacaiguoguogmail** [comment]
1.I rebase and squash to a single commit in local branch
2.then I `git pull`, lead to a merge commit
3.`git push`
How do I deal with this? @dnkoutso

File: spec/functional/command/ipc/podfile_json_spec.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+module Pod
+  describe Command::IPC::Podfile do
+    before do
+      Command::IPC::Podfile.any_instance.stubs(:output_pipe).returns(UI)
+    end
+
+    it 'converts a Podfile to JSON and prints it to STDOUT' do
+      out = run_command('ipc', 'podfile-json', fixture('Podfile'))
+      out.should == ''
```
Comment (dnkoutso) [inline spec/functional/command/ipc/podfile_json_spec.rb]: this cant be right, right?

**Comment by dnkoutso** [comment]
@dacaiguoguogmail not sure what failed in CircleCI, worth re-kicking the build. /cc @DanToml

**Comment by dacaiguoguogmail** [comment]
@dnkoutso I fixed the change log, triggered the build, still build failure in CircleCI

**Comment by orta** [comment]
Looks like your failure is unrelated to your PR - I can't re-trigger it alas

**Comment by dnkoutso** [comment]
Don't see a reason why this couldn't be merged.

**Pull request merged**